core/src/main/java/com/graphhopper/config/CustomArea.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.config;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.locationtech.jts.geom.Polygon;
+
+import com.graphhopper.routing.util.spatialrules.SpatialRule;
+
+/**
+ * Defines a named area which can be encoded in the graph and be targeted with a {@link SpatialRule}
+ * 
+ * @author Thomas Butz
+ */
+public class CustomArea {
```

Comment:
As this replaces the SpatialRules should we rename the PR to `make spatial rules customizable (CustomArea)`?

And can we somehow avoid the name clash with the other CustomArea class?

For future reference can you also list the renamed classes here as github somehow fails to catch all some of them. E.g. SpatialRuleLookupJTS, SpatialRuleLookupHelper (?) This would also help which classes I can skip when reviewing.

And why did you choose to keep some of the classes the same name like SpatialRuleSet? Could it make sense to rename it also to CustomAreaRules or something? Also the name for CountriesSpatialRuleFactory seems not consistent.

Reply:
> As this replaces the SpatialRules should we rename the PR to `make spatial rules customizable (CustomArea)`?

The SpatialRules are still there - at least until we fully replace them with CustomModels later on. The only thing that changed is that they don't provide their own border polygons anymore. SpatialRules only get applied now if their ID matches the one of a CustomArea.

> And can we somehow avoid the name clash with the other CustomArea class?

I think you're referring to `com.graphhopper.routing.ev.CustomArea` ? I can move the prefix and EV key creation into the regular CustomArea and delete this one.

> For future reference can you also list the renamed classes here as github somehow fails to catch all some of them. E.g. SpatialRuleLookupJTS, SpatialRuleLookupHelper (?) This would also help which classes I can skip when reviewing.
> 
> And why did you choose to keep some of the classes the same name like SpatialRuleSet? Could it make sense to rename it also to CustomAreaRules or something? Also the name for CountriesSpatialRuleFactory seems not consistent.

CustomAreaRule sounds good. I'm going to rename the remaining classes accordingly.